airflow/api_connexion/endpoints/task_instance_endpoint.py

```diff
@@ -545,3 +547,100 @@ def post_set_task_instances_state(*, dag_id: str, session: Session = NEW_SESSION
         session=session,
     )
     return task_instance_reference_collection_schema.dump(TaskInstanceReferenceCollection(task_instances=tis))
+
+
+@security.requires_access(
+    [
+        (permissions.ACTION_CAN_EDIT, permissions.RESOURCE_DAG),
+        (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG_RUN),
+        (permissions.ACTION_CAN_EDIT, permissions.RESOURCE_TASK_INSTANCE),
+    ],
+)
+@provide_session
+def patch_mapped_task_instance(
+    *, dag_id: str, dag_run_id: str, task_id: str, map_index: int, session: Session = NEW_SESSION
+) -> APIResponse:
+    """Update the state of a mapped task instance."""
```

Comment:
```suggestion
    """Update the state of a task instance."""
```
I think we shouldn't make this only for mapped task instance. Why do you think it should only be for mapped task?

Reply:
Thanks for feedback @ephraimbuddy  There is an other endpoint for the updating a simple task instance but @uranusjr suggested to de-duplicate them so am going to update it.

Reply:
Thanks for feedback @ephraimbuddy  There is an other endpoint for the updating a simple task instance but @uranusjr suggested to de-duplicate them so am going to update it.